### ISSUE TITLE
merge.unique problem

### ISSUE THREAD

**Issue by eightHundreds** [survivejs/webpack-merge#125]
``` js
const _ = require("lodash");
const merge = require("webpack-merge");

const PluginA = function(a) {
  this.a = a;
};
const PluginB = function() {};

console.log(
  merge({
    customizeArray: merge.unique(
      "plugins",
      ["PluginA"],
      plugin => plugin.constructor && plugin.constructor.name
    )
  })(
    {
      plugins: [new PluginB()]
    },
    {
      plugins: [new PluginA(), new PluginB()]
    }
  )
);
```

output
```
{ plugins: [ PluginB {}, PluginB {} ] }
```

expect result
```
{ plugins: [ PluginB {},PluginA {}, PluginB {} ] }
```

**Comment by InsOpDe** [comment]
Same problem here

**Comment by InsOpDe** [comment]
quick'n'dirty solution
```javascript
const merge = require("webpack-merge");
const _ = require("lodash");
const PluginA = function (a) {
	this.a = a;
};
const PluginB = function () { };
const PluginC = function () { };

console.log(
	merge({
		customizeArray(a, b, key) {
			if (key === 'plugins') {
				return [...b, ...a].reduce((sum, val)=>{
					let inside = false;
					for(const entry of sum){
						if(entry.constructor && val.constructor && val.constructor.name == "PluginA") {
							inside = inside || entry.constructor.name === val.constructor.name
						}
					}
					if (!inside) {
						sum.push(val);
					}
					return sum;
					}, []);
			}
			// Fall back to default merging
			return undefined;
		}
	})(
		{
			plugins: [new PluginB(), new PluginA(2)]
		},
		{
			plugins: [new PluginA(1), new PluginC(), new PluginB()]
		}
	)
);
```

**Comment by bebraw** [comment]
This will be fixed in #135 (v5).
